### PR TITLE
Podcasting widget

### PR DESCRIPTION
**Pull request by benlk**
## Changes

This pull request makes the following changes:

- creates widget for podcasts:
    - no thumbnail, just a plain-text play icon copied from Workday Minnesota work
    - choose by category
    - link in widget admin so users can get to the list of posts that are both in the chosen category and in the "Featured in Category" prominence term
    - includes styles for an optional "More Link" button at the bottom of the widget
- minor modifications to `partials/widget-content.php` to support the play icon
- updates `.btn` and `.btn-primary` styles in a direction more in line with the interactions mockup

Screen Shot 2019-10-24 at 17 44 17
Screen Shot 2019-10-24 at 17 44 25
Screen Shot 2019-10-24 at 19 17 43
Screen Shot 2019-10-24 at 19 17 48
Screen Shot 2019-10-24 at 19 18 04
Screen Shot 2019-10-24 at 19 18 29

## Why

For #45 

## Testing/Questions

Features that this PR affects:

- wodgets

Questions that need to be answered before merging:

- [x] Is this PR targeting the correct branch in this repository?
- [x] Works?
- [x] Controls make sense?
- [ ] for smaller display on mobile, would we want to align the podcast play button beside the headline?

Steps to test this PR:

1. Add a "Citylimits Podcasts Widget" to the "Homepage Bottom" widget area
2. Configure it to pull from a category
3. Verify that no posts in that category have the "Featured in Category" prominence term
4. Check that the output matches up: should be the three most-recent posts in the category
5. Add an older post in the category to that prominence term
6. Check that the output matches up: that post should appear, and then the rest of the posts in the widget should be backfilled
7. Adjust the count up and down by 3 to make sure styles look okay with n > 3 posts.

### CODE REVIEW

File: wp-content/themes/citylimits/less/widget-citylimits-podcast.less [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    line-height: @circlesize;
+    clear: both;
+    margin-left: auto;
+    margin-right: auto;
+
+    padding-left: 0;
+    padding-right: 0;
+    margin-bottom: @measure;
+    a {
+      display: tnline-block;
```
Comment (joshdarby) [inline wp-content/themes/citylimits/less/widget-citylimits-podcast.less]: @benlk Should be `inline-block`

**Comment by joshdarby** [comment]
@benlk For some reason this branch prevents everything after the `City Limits Podcasts` widget on `widgets.php` from loading

Screen Shot 2019-10-28 at 10 03 51 AM

**Comment by benlk** [comment]
> PHP message: PHP Recoverable fatal error:  Object of class WP_Error could not be converted to string in /wp-content/themes/citylimits/inc/widgets/class-citylimits-podcast-widget.php on line 259

**Comment by benlk** [comment]
Added handling for the state where the widget doesn't have a saved category:

Screen Shot 2019-10-28 at 12 33 09
Screen Shot 2019-10-28 at 12 33 18

File: wp-content/themes/citylimits/less/widget-citylimits-podcast.less (hunk 1 above)
Comment (benlk) [reply wp-content/themes/citylimits/less/widget-citylimits-podcast.less]: Fixed in 82dd089

**Comment by joshdarby** [comment]
@benlk I think this works well and looks good. I know the design only has 1 row of podcasts, so this can be up to interpretation, but do you think we should add borders below each row of podcasts if multiple rows exist like pictured in the screenshots above?

**Comment by benlk** [comment]
I'm not sure how we'd do that in a CSS Grid implicit layout? We'd need something like a `:nth-child( n>3)` selector.

File: wp-content/themes/citylimits/less/widget-citylimits-podcast.less (hunk 1 above)
Comment (benlk) [reply wp-content/themes/citylimits/less/widget-citylimits-podcast.less]: Yeah, it doesn't make sense for `.mediaplay` to have `cursor:pointer` and color changes within `.encircle` if `.mediaplay` is so much larger than the contained `.encircle` of the play icon, and invisible. Changed in 45e2811

**Comment by joshdarby** [comment]
@benlk We could probably do something like 
```
li:first-child:nth-last-child(n + 4), li:first-child:nth-last-child(n + 4) ~ li {
}
```

copied from here: https://css-tricks.com/solved-with-css-logical-styling-based-on-the-number-of-given-elements/

**Comment by benlk** [comment]
Okay, got that to work.

**Pull request merged**